test.functional/build.gradle.kts

```diff
@@ -1,7 +1,62 @@
+import com.moowork.gradle.node.yarn.YarnTask
+import java.util.function.Consumer
+
 plugins {
     id("java")
+    id("com.cognifide.aem.instance")
+    id("com.moowork.node")
 }
 
 description = "Example - Functional Tests"
 
+tasks{
+    register<YarnTask>("testFunctional") {
+        dependsOn("testSetup", "testInstallNode")
+        group = "check"
+        setYarnCommand("jest")
+        setWorkingDir(file("${projectDir.absolutePath}/test.functional"))
+        outputs.dir(file("${projectDir.absolutePath}/test.functional"))
+        doFirst {
+            val props = mutableListOf()
+            if (project.hasProperty("aem.env")) {
+                props.addAll(listOf("--config", "./env/${project.properties["aem.env"]}.config.js"))
```

Comment:
in groovy DSL accessing full map `project.properties` was wrong because it causes evaluation of all lazy properties tasks and extensions. to be safe I prefer project.findProperty() 

but... you have ready to consume `aem.environment` accessible here. your version is not null safe ("null" as string could be here)

Reply:
Done